presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcPageSource.java

```diff
@@ -243,13 +222,18 @@ private void closeWithSuppression(Throwable throwable)
         }
     }
 
-    private static Block buildSequenceBlock(long start, int count)
+    private static Block buildSequenceBlock(long start, int count, RowsToKeepResult rowsToKeep)
```

Comment:
For this function, we don't need to build a seq block then mask it. Just directly create the values based on `rowsToKeep`